scripts/install-cni.sh

```diff
@@ -1,27 +1,53 @@
-#!/bin/sh
-
+#! /bin/bash
 set -e -x;
 
-if [ -w "/host/opt/cni/bin/" ]; then
-    cp -f /install/bin/* /host/opt/cni/bin/;
-    echo "Wrote CNI binaries to /host/opt/cni/bin/";
+CONF_FILE_CONTAINER_PATH="/install/config"
+CONF_FILE_HOST_PATH="/host/etc/cni/net.d/"
+BIN_FILE_CONTAINER_PATH="/install/bin"
+BIN_FILE_HOST_PATH="/host/opt/cni/bin/"
+CONF_FILE="${CONF_FILE_CONTAINER_PATH}/..data/${CONF_FILE_NAME}"
+
+# Validating CIDR from network configuration file
+checksubnet() {
+    SUBNET=$(cat ${CONF_FILE} | jq 'select(.ipam.subnet != null) | .ipam.subnet' | tr -d \")
+    # If 'subnet' key it self not provided in network conf
+    if [ -z "${SUBNET}" ] ; then
+        return 0
+    fi
+    if [ -z $(egrep '^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(\/([0-9]|[1-2][0-9]|3[0-2]))$' <<< $SUBNET) ]; then
+        echo "Invalid CIDR mentioned in the conf file ${CONF_FILE_NAME}"
+  	    if [ $1 -eq 1 ]; then # exit the script
+  	        exit 255
+  	    fi
+        return 1
+    fi
+        return 0
```

Comment:
indentation Ajey.

There is no else block. why 'return 0' is starting from there

Reply:
done

Reply:
yes,  lets not ignore indentation please. makes it more readable.